services/121-service/src/user/user.service.ts

```diff
@@ -271,18 +276,26 @@ export class UserService {
         UserType.aidWorker,
       );
 
-      const emailPayload: CreateUserEmailPayload = {
-        email: userEntity.username ?? '',
-        displayName: userEntity.displayName ?? '',
+      if (!userEntity.username) {
+        throw new Error('Invariant violated: userEntity.username is missing');
```

Comment:
```suggestion
        throw new Error('Username is missing');
```

I think this error could be returned to the end-user, so it doesn't seem very 'understandable' for 'an avarage user'?

(Also not even to me. :P )

Also, wouldn't this be covered by a 'validator' on the DTO (because this property would be `required`?)